### PR TITLE
jq: remove bison requirement plus fix `brew audit` spacing nit

### PR DESCRIPTION
**Pull request by Habbie**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
### Description

Please refer to https://github.com/stedolan/jq/blob/master/configure.ac#L38 to see the Bison requirement.

### CODE REVIEW

File: Formula/jq.rb [hunk 1]
```diff
@@ -21,8 +21,8 @@ class Jq < Formula
     depends_on "libtool" => :build
   end
 
-  depends_on "oniguruma"  # jq depends > 1.5
-  depends_on "bison" => :build # jq depends on bison > 2.5
+  depends_on "oniguruma" # jq depends > 1.5
+  depends_on "bison" => :build # jq depends on bison >= 3.0
```
Comment (rakkesh) [inline Formula/jq.rb]: `bison` dependency can be removed if you could add `--disable-maintainer-mode` to configure options.

```rb
    system "./configure", "--disable-dependency-tracking",
                          "--disable-silent-rules",
                          "--disable-maintainer-mode",
                          "--prefix=#{prefix}"
```

Comment (Habbie) [inline Formula/jq.rb]: That would also make sense, as we are just building from a release tarball. I'm happy to change the PR to do that instead, would save a dependency.

Comment (DomT4) [inline Formula/jq.rb]: Seems a reasonable suggestion.

Comment (Habbie) [inline Formula/jq.rb]: Ok, I'll do that.

**Comment by Habbie** [comment]
Updated as discussed.

**Comment by UniqMartin** [comment]
Merged in 85b2c055cc858e17b952744c2c3e6a98fd0232f9. Thank you for this contribution to Homebrew, @Habbie! :tada:

**Pull request closed without merging**